src/api-client.ts

```diff
@@ -44,13 +44,31 @@ export abstract class ApiClient {
     return paramString ? `${pathUrl}?${paramString}` : pathUrl;
   }
 
+  fetchEntity(
```

Comment:
Not sure on the naming of fetchEntity (entity is an overloaded term). I like what you did with "fetchBlob" as that is exactly what the method returns. Maybe something like "fetchJson" as it's closer to what the method does?

Reply:
fetchObject?